### ISSUE TITLE
sql: panic: caused by hashjoiner-limited: unexpected 10240 leftover bytes

### ISSUE THREAD

**Issue by mjibson** [cockroachdb/cockroach#34307]
Sadly the SQL statement is truncated.

```
E190128 22:10:56.249261 515 sql/conn_executor.go:697  [n1,client=127.0.0.1:34824,user=root] a SQL panic has occurred while executing "INSERT INTO table1(col0, col1, col2) ((VALUES ((SELECT NULL FROM table0 AS tab_2 LIMIT 1), B'10100111101111111111111011', NULL), (NULL, B'010011000110110101101011001011010000110001001101111111100111000101011011010000111010010000111111', time_in('20:15:02.492558')), ('8', B'11010111001101100111000001101001011101000101111100010100111100000011100000000', '03:08:36.448069'), ('', CASE WHEN false THEN B'01010000010100100101101000001001100011001101010011101010' ELSE B'0111111101' END, '15:19:29.802565 [...]": hashjoiner-limited: unexpected 10240 leftover bytes
E190128 22:10:56.249493 515 util/log/crash_reporting.go:202  [n1,client=127.0.0.1:34824,user=root] a panic has occurred!
E190128 22:10:56.255686 515 util/log/crash_reporting.go:476  [n1,client=127.0.0.1:34824,user=root] Reported as error 26e4bc8d1b194755b736f89c95f3540f
panic: hashjoiner-limited: unexpected 10240 leftover bytes [recovered]
	panic: panic while executing 1 statements: INSERT INTO _(_, _, _) ((VALUES ((SELECT _ FROM _ AS _ LIMIT _), _, _), (__more3__)) UNION ALL (VALUES (_, _, CASE WHEN _ THEN _ ELSE _ END), (__more1__))) INTERSECT ALL (VALUES (_, _, __more1__), (__more1__)); caused by hashjoiner-limited: unexpected 10240 leftover bytes

goroutine 515 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc00172c000, 0x2ed9bc0, 0xc00120b1c0, 0x243fb40, 0xc0002f1e50)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:711 +0x36d
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1(0xc00172c000, 0x2ed9bc0, 0xc00120b1c0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:423 +0x61
panic(0x243fb40, 0xc0002f1e50)
	/usr/local/go/src/runtime/panic.go:513 +0x1b9
github.com/cockroachdb/cockroach/pkg/util/log.ReportOrPanic(0x2ed9c80, 0xc0016ab740, 0xc0004b5300, 0xc0013b4240, 0x33, 0x0, 0x0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/util/log/crash_reporting.go:493 +0x207
github.com/cockroachdb/cockroach/pkg/util/mon.(*BytesMonitor).doStop(0xc000d25c20, 0x2ed9c80, 0xc0016ab740, 0xc0016ab701)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/util/mon/bytes_usage.go:397 +0x25d
github.com/cockroachdb/cockroach/pkg/util/mon.(*BytesMonitor).Stop(0xc000d25c20, 0x2ed9c80, 0xc0016ab740)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/util/mon/bytes_usage.go:383 +0x44
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*hashJoiner).close(0xc001094d80)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/hashjoiner.go:611 +0xe7
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.newHashJoiner.func1(0x2ed9c80, 0xc0016ab740, 0x0, 0x0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/hashjoiner.go:167 +0x2a
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*ProcessorBase).moveToTrailingMeta(0xc001094d80)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/processors.go:764 +0x132
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*ProcessorBase).DrainHelper(0xc001094d80, 0x5f0fb8)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/processors.go:698 +0x1bd
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*hashJoiner).Next(0xc001094d80, 0xc001306420, 0xc001ba5900, 0xc001b95e98, 0x627e61)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/hashjoiner.go:260 +0x2ae
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*ProcessorBase).DrainHelper(0xc001bc8f00, 0xc001ba59f0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/processors.go:694 +0x9e
github.com/cockroachdb/cockroach/pkg/sql.(*planNodeToRowSource).Next(0xc001bc8f00, 0x0, 0x0, 0x0, 0xc001ba59f0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/plan_node_to_row_source.go:207 +0x6dc
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.Run(0x2ed9c80, 0xc0016ab740, 0x2ee3240, 0xc001bc8f00, 0x2ecf100, 0xc001bce380)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/base.go:174 +0x35
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*ProcessorBase).Run(0xc001bc8f00, 0x2ed9c80, 0xc0016ab740)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/processors.go:802 +0x92
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*Flow).Run(0xc001921e00, 0x2ed9c80, 0xc0016ab740, 0x29380d0, 0x0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/flow.go:657 +0x209
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run(0xc000466c80, 0xc0010354a0, 0xc00188a1b0, 0xc001b969a0, 0xc0011462c0, 0xc00172c518, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:251 +0x8a0
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun(0xc000466c80, 0x2ed9c80, 0xc000fb0b70, 0xc00172c518, 0xc0010354a0, 0xc00188a1b0, 0x2edb480, 0xc001306420, 0xc0011462c0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:794 +0x200
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine(0xc00172c000, 0x2ed9c80, 0xc000fb0b70, 0xc00172c450, 0x2, 0x7f1f22f81528, 0xc00188a240, 0x0, 0x0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1044 +0x26e
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc00172c000, 0x2ed9c80, 0xc000fb0b70, 0xc00172c450, 0x7f1f22f81528, 0xc00188a240, 0x0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:899 +0x644
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0xc00172c000, 0x2ed9c80, 0xc000fb0b70, 0x2edec00, 0xc001ba5220, 0xc0018e44e9, 0x388, 0x0, 0x0, 0x0, ...)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:437 +0xdd9
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc00172c000, 0x2ed9c80, 0xc000fb0b70, 0x2edec00, 0xc001ba5220, 0xc0018e44e9, 0x388, 0x0, 0x0, 0x0, ...)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:93 +0x36f
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc00172c000, 0x2ed9bc0, 0xc00120b1c0, 0xc000113bf8, 0x5400, 0x15000, 0xc000113c90, 0xc00167b640, 0x0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1172 +0x21e5
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc0005d6b60, 0x2ed9bc0, 0xc00120b1c0, 0xc00172c000, 0x5400, 0x15000, 0xc000113c90, 0xc00167b640, 0x0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:425 +0xce
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).serveImpl.func4(0xc0005d6b60, 0x2ed9bc0, 0xc00120b1c0, 0xc00172c000, 0x5400, 0x15000, 0xc000113c90, 0xc00167b640, 0xc00167b650, 0xc000481374)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:319 +0x81
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).serveImpl
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:318 +0x1033
```

tables created with:

```
CREATE TABLE table0 (col0 UUID NOT NULL, col1 UUID NOT NULL, col2 FLOAT8, col3 VARBIT NULL, col4 INET NULL, col5 BYTES NOT NULL, col6 OID NULL, col7 TIMESTAMP, col8 UUID, col9 BOOL NULL, col10 NAME NULL, col11 BYTES NULL, FAMILY fam0 (col11, col2), FAMILY fam1 (col3, col5, col8, col1, col7), FAMILY fam2 (col6, col10, col0), FAMILY fam3 (col4, col9), UNIQUE (col7, col3, col10, col1, col5 DESC, col9 DESC, col11 ASC), INDEX (col4 ASC));
CREATE TABLE table1 (col0 NAME, col1 VARBIT NOT NULL, col2 TIME NULL, FAMILY fam0 (col1, col2, col0), PRIMARY KEY (col2 DESC), UNIQUE (col0 DESC));
```

Found with sqlsmith. Assigning for triage.

**Comment by jordanlewis** [comment]
Oh no! If you can find a reproduction of this without the truncated query, it would be invaluable. Bugs like this are really hard to reproduce.

**Comment by yuzefovich** [comment]
I feel like #34092 might be the reason.

**Comment by mjibson** [comment]
[smither.txt](https://github.com/cockroachdb/cockroach/files/2806370/smither.txt)

I've attached a SQL log here of another reproduction. You will have to...do a thing to modify it so that it ignores statements that produce an error. Like probably write a script that just executes `cockroach sql -e` for each line of the file and doesn't stop on error. Boa sorte.

**Comment by jordanlewis** [comment]
@yuzefovich, please attempt to determine if it's fixed by your latest patch and close if so.

**Comment by yuzefovich** [comment]
I believe it is the same issue as #34405 caused by a bug in #34092 which is now fixed by #34407 , so I'm closing this issue.
